src/main/java/pack/controller/PersonController.java

```diff
@@ -1,102 +1,88 @@
 		request = new HttpGet("https://jsonplaceholder.typicode.com/users");
 		objectMapper = new ObjectMapper();
-		personsList=new ArrayList<>();
+		node = factory.objectNode();
+		child = factory.objectNode();
 	}
 
-	@RequestMapping(value = "/getPersons", method = RequestMethod.GET)
-	@Override
+	@RequestMapping(value = "/getPersons", method = RequestMethod.GET, produces = "application/json")
 	@ResponseBody
-	public List<Person> getPersons() {
-		return personService.getPersons();
+	public ObjectNode getPersons() throws CustomException, JsonProcessingException {
+
+		try {
+			String json = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(personService.getPersons());
+			child.put("status", HttpStatus.ACCEPTED.getReasonPhrase());
+			child.put("data", json);
+			node.set("result", child);
+			return node;
+		} catch (CustomException e) {
+			throw new CustomException();
+		}
 	}
 
-	@Override
 	@RequestMapping(value = "/findPersonById/{id}", method = RequestMethod.GET)
 	@ResponseBody
 	public Person findPersonById(@PathVariable("id") int id) {
-		return personService.findPersonById(id);
-	}
-
-	@Override
-	@RequestMapping(value = "/savePersonListFromJsonplaceholder", method = RequestMethod.POST)
-	@ResponseBody
-	public boolean savePersonListFromJsonplaceholder() {
-		Person pers=new Person();
-		for (Person p : getPersonsFromJsonplaceholder()) {
-			
-			pers.setName(p.getName());
-			pers.setUsername(p.getUsername());
-			pers.setEmail(p.getEmail());
-			pers.setPhone(p.getPhone());
-			pers.setWebsite(p.getWebsite());
-		//	pers.setAddress(new Address(p.getAddress().getStreet(), p.getAddress().getSuite(), p.getAddress().getCity(), p.getAddress().getZipcode(), p, p.getAddress().getGeo()));
-			//pers.setCompany(new Company(pers.getCompany().getName(), pers.getCompany().getCatchPhrase(), pers.getCompany().getBs(), p));
-		
-			pers.setAddress(p.getAddress());
-			pers.setCompany(p.getCompany());
-			personsList.add(pers);
+		try {
+			return personService.findPersonById(id);
+		} catch (CustomException e) {
+			e.printStackTrace();
+			return new Person();
 		}
-		
-		return personService.savePersonListFromJsonplaceholder(personsList);
 	}
 
-	@Override
-	@RequestMapping(value = "/getPersonsFromJsonplaceholder", method = RequestMethod.GET)
+	@RequestMapping(value = "/getPersonsFromJsonplaceholder", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON)
 	@ResponseBody
 	public List<Person> getPersonsFromJsonplaceholder() {
 		try {
 			response = httpClient.execute(request);
 			String result = EntityUtils.toString(response.getEntity());
 			return objectMapper.readValue(result,
-					objectMapper.getTypeFactory().constructCollectionType(List.class, Person.class));
 
-		} catch (IOException e) {
+					objectMapper.getTypeFactory().constructCollectionType(List.class, Person.class));
+		} catch (Exception e) {
 			e.printStackTrace();
 			return Collections.emptyList();
```

Comment:
Same comments as above. I don't want for the server to return a default 500 http status. I want a custom data object with no data, a custom http status code and a reason for the failure.